### PR TITLE
Networking mirror

### PR DESCRIPTION
**Pull request by eric-zdw**
This is a big change, with a lot of code rewritten to fit networking code.

Networking code uses the [Mirror library](https://mirror-networking.com/docs/).
Comments were added to PlayerBehaviour for some explanation on Mirror code; check the docs for more information.

### CODE REVIEW

File: Assets/NewNetworkManager.cs [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    /// <summary>
+    /// Called on clients when disconnected from a server.
+    /// <para>This is called on the client when it disconnects from the server. Override this function to decide what happens when the client disconnects.</para>
+    /// </summary>
+    /// <param name="conn">Connection to the server.</param>
+    public override void OnClientDisconnect(NetworkConnection conn)
+    {
+        base.OnClientDisconnect(conn);
```
Comment (Humeian) [inline Assets/NewNetworkManager.cs:204]: There should probably be some code here to reset player1 & player2 if a client disconnects for whatever reason

**Comment by Humeian** [comment]
Generally looks good to me, haven't had a chance to test it yet though. Did you modify any of the Mirror library files? I didn't actually look at those

**Comment by eric-zdw** [comment]
> Generally looks good to me, haven't had a chance to test it yet though. Did you modify any of the Mirror library files? I didn't actually look at those

No core Mirror library files were changed, though NewNetworkManager overrides some events from NetworkManager.

**Comment by eric-zdw** [comment]
Will merge later this afternoon if there are no objections.

**Pull request merged**